### PR TITLE
SLVS-1242 Initialize Telemetry

### PR DESCRIPTION
**Pull request by vnaskos-sonar**

### CODE REVIEW

File: src/Integration/SLCore/SLCoreConstantsProvider.cs [hunk 1]
```diff
@@ -33,9 +35,10 @@ namespace SonarLint.VisualStudio.Integration.SLCore
...
     {
         private readonly Lazy<string> ideName;
+        private readonly IVsVersion ideVersion;
 
         [ImportingConstructor]
-        public SLCoreConstantsProvider(IVsUIServiceOperation vsUiServiceOperation)
+        public SLCoreConstantsProvider(IVsUIServiceOperation vsUiServiceOperation, IVsVersionProvider vsVersionProvider)
         {
             // lazy is used to keep the mef-constructor free threaded and to avoid calling this multiple times
             ideName = new Lazy<string>(() =>
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration/SLCore/SLCoreConstantsProvider.cs:44]: Could this be moved to the IVsVersionProvider? Potentially with a rename. The Ide name and the Ide version seem to be related enough to be placed in the same class

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]

File: src/Integration/SLCore/SLCoreConstantsProvider.cs (hunk 1 above)
Comment (vnaskos-sonar) [reply src/Integration/SLCore/SLCoreConstantsProvider.cs:44]: https://sonarsource.atlassian.net/browse/SLVS-1258

File: src/Integration/SLCore/SLCoreConstantsProvider.cs [hunk 2]
```diff
@@ -72,5 +78,20 @@ public SLCoreConstantsProvider(IVsUIServiceOperation vsUiServiceOperation)
...
         ];
+        
+        private static IdeVersionInformation GetVsVersion(IVsVersion vsVersion)
+        {
+            if (vsVersion == null)
+            {
+                return null;
+            }
+
+            return new IdeVersionInformation
```
[the remark this answers is not in the document]
Comment (vnaskos-sonar) [reply src/Integration/SLCore/SLCoreConstantsProvider.cs:89]: Already covered on the `SLCoreConstantsProviderTests.cs`

File: src/Integration.UnitTests/SLCore/SLCoreConstantsProviderTests.cs [hunk 3]
```diff
@@ -28,145 +31,188 @@
...
+        version.DisplayVersion.Returns("17.10.0 Preview 3.0");
+        versionProvider.Version.Returns(version);
+
+        var testSubject = CreateTestSubject(versionProvider: versionProvider);
+        VisualStudioHelpers.VisualStudioVersion = "1.2.3.4";
+        var expectedString = $$"""
+                         {
+                           "productKey": "visualstudio",
+                           "productName": "SonarLint Visual Studio",
+                           "productVersion": "{{typeof(TelemetryTimer).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>().Version}}",
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.UnitTests/SLCore/SLCoreConstantsProviderTests.cs:105]: `TelemetryTimer` is going to be deleted, use a different class

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]

File: src/Integration.UnitTests/SLCore/SLCoreConstantsProviderTests.cs (hunk 3 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.UnitTests/SLCore/SLCoreConstantsProviderTests.cs:105]: https://github.com/SonarSource/sonarlint-visualstudio/pull/5531/files#diff-1981c848372dfb6ebd8e3346e790c002216a833fac2332807552c9feeab9237dR36

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Pull request merged**